waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1483,16 +1684,253 @@ public CacheFileMetadataResult cache_file_metadata(CacheFileMetadataRequest req)
+    List<SchemaVersion> result = databaseMapping.getClient().get_schema_all_versions(iSchemaName);
+    return databaseMapping.transformOutboundSchemaVersions(result);
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public void drop_schema_version(SchemaVersionDescriptor schemaVersionDescriptor) throws NoSuchObjectException, MetaException, TException {
+    DatabaseMapping databaseMapping = checkWritePermissions(schemaVersionDescriptor.getSchema().getDbName());
+    databaseMapping.getClient().drop_schema_version(schemaVersionDescriptor);
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public FindSchemasByColsResp get_schemas_by_cols(FindSchemasByColsRqst findSchemasByColsRqst) throws MetaException, TException {
+    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
+    FindSchemasByColsResp result = databaseMapping.getClient().get_schemas_by_cols(findSchemasByColsRqst);
+    return result;
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public void map_schema_version_to_serde(MapSchemaVersionToSerdeRequest mapSchemaVersionToSerdeRequest) throws NoSuchObjectException, MetaException, TException {
+    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
+    databaseMapping.getClient().map_schema_version_to_serde(mapSchemaVersionToSerdeRequest);
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public void set_schema_version_state(SetSchemaVersionStateRequest setSchemaVersionStateRequest) throws NoSuchObjectException, InvalidOperationException, MetaException, TException {
+    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
+    databaseMapping.getClient().set_schema_version_state(setSchemaVersionStateRequest);
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public void add_serde(SerDeInfo serDeInfo) throws AlreadyExistsException, MetaException, TException {
+    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
+    databaseMapping.getClient().add_serde(serDeInfo);
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public SerDeInfo get_serde(GetSerdeRequest getSerdeRequest) throws NoSuchObjectException, MetaException, TException {
+    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
+    SerDeInfo result = databaseMapping.getClient().get_serde(getSerdeRequest);
+    return result;
+  }
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public LockResponse get_lock_materialization_rebuild(String dbName, String tableName, long txnId) throws TException {
```

Comment:
Should we sent the request to the correct client based on the dbName?

Reply:
@Override
  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
  public LockResponse get_lock_materialization_rebuild(String dbName, String tableName, long txnId) throws TException {
    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
    LockResponse result = databaseMapping.getClient().get_lock_materialization_rebuild(dbName, tableName, txnId);
    return result;
  }

It's a override method. It's expecting dbName. Any suggestions? pls

Reply:
Check other methods where we use s dbName we need to transform it:
```
 DatabaseMapping mapping = databaseMappingService.databaseMapping(dbname);
mapping.transformInboundDatabaseName(dbname)
```
and also sent the call to the correct client.

Please check this for all methods that are added. Sometimes the database name is inside a complex object.

Reply:
Hi @patduin ,
Based on ur suggestion, modified the code. Pls check it

Reply:
Transformed the database name present in the request & response of complex object too

Reply:
thanks looking good, left some comments, almost there!